pkg/client/client_v2_test.go

```diff
@@ -869,6 +875,149 @@ func TestClientV2_Checkin_FeatureFlags(t *testing.T) {
 	require.Empty(t, errs)
 }
 
+type unitChangesAccumulator struct {
+	mx          sync.Mutex
+	unitChanges []UnitChanged
+}
+
+func (uca *unitChangesAccumulator) Start(ctx context.Context, t *testing.T, c <-chan UnitChanged) {
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case uc := <-c:
+				func() {
+					t.Logf("received unit change %v", uc)
+					switch uc.Type {
+					case UnitChangedAdded:
+						err := uc.Unit.UpdateState(UnitStateHealthy, "Healthy", map[string]interface{}{
+							"custom": "payload",
+						})
+						require.NoError(t, err)
+					}
+
+					uca.mx.Lock()
+					defer uca.mx.Unlock()
+
+					uca.unitChanges = append(uca.unitChanges, uc)
+				}()
+			}
+		}
+	}()
+}
+
+func (uca *unitChangesAccumulator) GetAccumulatedChanges() []UnitChanged {
+	uca.mx.Lock()
+	defer uca.mx.Unlock()
+
+	ret := make([]UnitChanged, len(uca.unitChanges))
+	copy(ret, uca.unitChanges)
```

Comment:
Not sure how much this matters but if `uca.unitChanges` is `nil`, `ret` will not be `nil` but instead be a slice of length 0. See https://go.dev/play/p/jzJ9cXkR6pZ.

Reply:
It does not matter too much here since we are going to check the length and contents and in both cases we have 0 length (so no content)
I don't think that the extra allocation for the zero-length slice is gonna cost us here...